NetflixClone/Classes/Scenes/Home/HomeViewController.swift

```diff
@@ -6,125 +6,192 @@
-        let item = NSCollectionLayoutItem(
-            layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(0.5),
-                                               heightDimension: .fractionalHeight(1))
+    private func configureHierarchy() {
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: createLayout())
+        collectionView.delegate = self
+        collectionView.register(
+            PosterCollectionViewCell.self,
+            forCellWithReuseIdentifier: PosterCollectionViewCell.reuseIdentifier
         )
-        
-        // Create Group
-        let group = NSCollectionLayoutGroup.horizontal(
-            layoutSize: NSCollectionLayoutSize(widthDimension: .fractionalWidth(1),
-                                               heightDimension: .absolute(150)), subitems: [item])
-        
-        //Create Section
-        let section = NSCollectionLayoutSection(group: group)
-        section.orthogonalScrollingBehavior = .paging
-        
-        // Create Supplementary Item
-        let headerItemSize = NSCollectionLayoutSize(widthDimension: .fractionalWidth(1), heightDimension: .estimated(100))
-        let headerItem = NSCollectionLayoutBoundarySupplementaryItem(layoutSize: headerItemSize, elementKind: "header", alignment: .top)
-        section.boundarySupplementaryItems = [headerItem]
-        
-        
-        return UICollectionViewCompositionalLayout(section: section)
+        collectionView.frame = view.bounds
+        collectionView.backgroundColor = .white
+        view.addSubview(collectionView)
+    }
+    private func createSectionHeaderRegistration() -> UICollectionView.SupplementaryRegistration<UICollectionViewListCell> {
+        return UICollectionView.SupplementaryRegistration<UICollectionViewListCell>(
+            elementKind: sectionHeaderElementKind
+        ) { [weak self] supplementaryView, _, indexPath in
+            guard let self = self else { return }
+            let sectionID = self.dataSource.snapshot().sectionIdentifiers[indexPath.section]
+            supplementaryView.configurationUpdateHandler = { supplementaryView, state in
+                guard let supplementaryCell = supplementaryView as? UICollectionViewListCell else { return }
+                if sectionID == .featured { return }
+                var contentConfiguration = UIListContentConfiguration.plainHeader().updated(for: state)
+                contentConfiguration.text = sectionID.rawValue
+                contentConfiguration.textProperties.color = UIColor.black
+                contentConfiguration.textProperties.font = UIFont.boldSystemFont(ofSize: 19)
+                supplementaryCell.contentConfiguration = contentConfiguration
+                supplementaryCell.backgroundConfiguration = .clear()
+            }
+        }
+    }
+    private func configureDataSource() {
+        dataSource = UICollectionViewDiffableDataSource
+        <Section, Movie>(collectionView: collectionView) {
```

Comment:
```suggestion
    typealias DataSource = UICollectionViewDiffableDataSource<Section, Movie>
    
    private func configureDataSource() {
        dataSource = DataSource(collectionView: collectionView) {
```
Typealiasss are great to minimize long repetitive names.

Reply:
I'll add typaliases for any other UICollectionViewDiffableDataSource from now on. Much cleaner🔥! Thanks!